.github/workflows/style.yml

```diff
@@ -18,6 +18,16 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+
+      - name: Python Style Check
+        run: |
+          pip install ruff
+          ruff check ./api
```

Comment:
3. check Python style by running ruff linter and list all the violations if it fails